eth/api.go

```diff
@@ -660,10 +660,10 @@ func (s *PublicBlockChainAPI) doCall(args CallArgs, blockNr rpc.BlockNumber) (st
 		data:     common.FromHex(args.Data),
 	}
 	if msg.gas.Cmp(common.Big0) == 0 {
-		msg.gas = big.NewInt(50000000)
+		msg.gas = common.MaxBig
```

Comment:
If your call runs out of gas you can always supply more gas and/or lower gas price as arguments. This hard coded value is just a fallback when the user doesn't supply a gas amount. You can debate if a hard coded value is good but in this case it might be a sensible default value. If you need more just supply more gas as arguments.

At this time I think this is still a good default value, it gives a good balance between guarding against infinite loops while still allowing most default uses cases.